gap/attr.gi

```diff
@@ -174,6 +174,201 @@ function(D)
   return DIGRAPHS_ArticulationPointsBridgesStrongOrientation(D)[4];
 end);
 
+# Utility function to calculate the maximal independent sets (as BLists) of a
+# subgraph induced by removing a set of vertices.
+BindGlobal("DIGRAPHS_MaximalIndependentSetsSubtractedSet",
+function(I, subtracted_set, size_bound)
+  local induced_mis, temp, i;
+  # First remove all vertices in the subtracted set from each MIS
+  temp := List(I, i -> DifferenceBlist(i, subtracted_set));
+  induced_mis := [];
+  # Then remove any sets which are no longer maximal
+  # Sort in decreasing size
+  Sort(temp, {x, y} -> SizeBlist(x) > SizeBlist(y));
+  # Then check elements from back to front for if they are a subset
+  for i in temp do
+    if SizeBlist(i) <= size_bound and
+        ForAll(induced_mis, x -> not IsSubsetBlist(x, i)) then
+      Add(induced_mis, i);
+    fi;
+  od;
+  return induced_mis;
+end
+);
+
+BindGlobal("DIGRAPHS_ChromaticNumberLawler",
+function(D)
+  local n, vertices, subset_colours, s, S, i, I, subset_iter, x,
+  mis, subset_mis;
+  n := DigraphNrVertices(D);
+  vertices := List(DigraphVertices(D));
+  # Store all the Maximal Independent Sets, which can later be used for
+  # calculating the maximal independent sets of induced subgraphs.
+  mis := DigraphMaximalIndependentSets(D);
+  # Convert each MIS to a Blist
+  mis := List(mis, x -> BlistList(vertices, x));
+  # Store current best colouring for each subset
+  subset_colours := ListWithIdenticalEntries(2 ^ n, infinity);
```

Comment:
The ultimate reason for the crashes is that on the second run-through of the test file, your new algorithms are (incorrectly) being run when they shouldn't be run (see https://github.com/gap-system/gap/issues/4510 - the option value was being (in my opinion, wrongly) retained between different calls to `ChromaticNumber` after an error).

The reason that this is a problem is that a list with `2 ^ n` entries quickly becomes HUGE and crashes GAP when `n` is big enough (and it doesn't have to be that big).

Would it be easy to avoid having such a large list? If not, perhaps these functions could give an error - or at least a warning - if they are called with a digraph with 'too many' vertices (whatever that should be).